include/vicon/udp.h

```diff
@@ -427,7 +467,8 @@ class UDPClient
                 // Lock mutex
                 std::lock_guard<std::mutex> guard(m_ObjectMutex);
 
-                const auto objectData = reinterpret_cast<RawObjectData *>(&buffer[5]);
+                auto objectData = reinterpret_cast<RawObjectData *>(&buffer[5]);
+                objectData->objectName[23] = '\0'; // Make sure string is null-terminated
```

Comment:
If the strings the vicon system returns aren't actually properly null-terminated and we always add a null terminator to the end it might actually be better to copy these strings with ``std:::copy_n`` to avoid Windows whining about insecure functions etc

Reply:
Maybe... There is one place where I pass in a ``std::string`` from a call site so this would mean copying it to a ``char[24]`` first -- but I agree this is cleaner